horace_core/sqw/PixelData/@PixelData/append.m

```diff
@@ -0,0 +1,62 @@
+function obj = append(obj, pix)
+% Join the pixels in the given PixelData object to the end of this
+% object.
+%
+% The pixels to append must all be in memory and you cannot append pixels if
+% the inputted PixelData object has more pixels than are allowed in a single
+% page.
+%
+% Input
+% -----
+% pix    A PixelData object containing the pixels to append
+%
+if ~isa(pix, 'PixelData')
+    error('PIXELDATA:append', ...
+            'Input object must be of class ''PixelData''. Found ''%s''', ...
+            class(pix));
+end
+if isempty(pix)
+    return;
+end
+if pix.num_pixels > obj.max_page_size_
+    error('PIXELDATA:append', ...
+            ['Cannot append more pixels than allowed in a single page\n ' ...
+            'Found ''%i'' pixels, ''%i'' allowed.'], pix.num_pixels, ...
+            obj.max_page_size_);
+end
+
+if ~(obj.get_num_pages_() == obj.page_number_)
+    if obj.page_is_dirty_(obj.page_number_) && obj.dirty_page_edited_
+        obj.write_dirty_page_();
+    end
+    obj.load_page_(obj.get_num_pages_());
+end
+
+if isempty(obj)
+    obj.data_ = pix.data;
+    obj.set_page_dirty_(true);
+else
+    if obj.page_size < obj.max_page_size_
+        num_to_allocate_to_pg = obj.max_page_size_ - obj.page_size;
+        obj.data_ = horzcat(obj.data, pix.data(:, 1:num_to_allocate_to_pg));
+        obj.set_page_dirty_(true);
+
+        if num_to_allocate_to_pg ~= pix.num_pixels
+            obj.write_dirty_page_();
+
+            obj.data_ = pix.data(:, (num_to_allocate_to_pg + 1):end);
+            obj.page_number_ = obj.page_number_ + 1;
+        end
+
+    elseif obj.page_size == obj.max_page_size_
+        obj.set_page_dirty_(true);
```

Comment:
`obj.set_page_dirty_(true);` appears in all three branches of the `if` -- is there anything preventing that being moved to before the start of the `if` (is order important)

Reply:
I've cleaned up the logic a bit - but I don't think it can be brought outside. Order is important here, in some branches we edit an existing page (so it must be marked dirty) then we create a new page, which must also be marked dirty. If the original page is full, then we don't edit it so it need not be marked dirty.